include/boost/http_proto/serializer.hpp

```diff
@@ -297,44 +325,130 @@ class BOOST_SYMBOL_VISIBLE
 
 //------------------------------------------------
 
+/**
+    A proxy type used to pass bounded input to the
+    associated serializer.
```

Comment:
No one cares that a type is "a proxy type." And if we refactor the implementation (which should be invisible to the user) to not be "a proxy type" this would invalidate the javadoc. When writing a brief or javadoc description it is best to speak in general terms which will be relevant regardless of any changes in implementation.

Also, "bounded input" is not particularly descriptive as it is not really relevant. The distinguishing characteristic of `serializer::stream` is that the `serializer` is not on the call stack at the moment that the caller provides a buffer representing part of the body.

Suggestion:
"The type used for caller-provided body data during serialization."